File: Person.java
public class Person {
    private static int ID = 0;
    private String name = "";
    private String lastName = "";
    private String birthday = "";
    private String CURP = "";
    private int numVisits = 0;
    private String dateOfRegister = "";

    public Person(String name, String lastName, String birthday, String CURP, int numVisits, String dateOfRegister) {
        this.name = name;
        this.lastName = lastName;
        this.birthday = birthday;
        this.CURP = CURP;
        this.numVisits = numVisits;
        this.dateOfRegister = dateOfRegister;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public void setBirthday(String birthday) {
        this.birthday = birthday;
    }

    public void setCURP(String CURP) {
        this.CURP = CURP;
    }

    public void addVisit() {
        numVisits++;
    }

    public void setDateOfRegister(String dateOfRegister) {
        this.dateOfRegister = dateOfRegister;
    }

    public static int getID() {
        return ID;
    }

    public String getName() {
        return String.format("%s %s",name,lastName);
    }

    public String getBirthday() {
        return birthday;
    }

    public String getCURP() {
        return CURP;
    }

    public int getNumVisits() {
        return numVisits;
    }

    public String getDateOfRegister() {
        return dateOfRegister;
    }
}

File: Person.py
import uuid

class Person:
    __id = " "
    __name = " "
    __last_name = " "
    __birthday = " "
    __curp = " "
    __num_visits = 0
    __date_of_register = " "

    def __init__(self, name, last_name, birthday, curp, date_of_register):
        self.__name = name
        self.__last_name = last_name
        self.__birthday = birthday
        self.__curp = curp
        self.__date_of_register = date_of_register
        self.__set_id()

    def __set_name(self, name, last_name):
        self.__name = name
        self.__last_name = last_name

    def __set_birthday(self, birthday):
        self.__birthday = birthday
        self.__set_id()

    def __set_curp(self, curp):
        self.__curp = curp

    def __set_num_visits(self, num_visits):
        self.__num_visits = num_visits

    def __add_visit(self):
        __num_visits = __num_visits + 1

    def __set_date_of_register(self, date_of_register):
        self.__date_of_register = date_of_register

    def __get_id(self):
        return self.__id
    
    def __set_id(self):
        year = 0
        print(self.__birthday)
        year = int(self.__birthday[-4:])
        if 2024 - year > 17 : 
            xd = uuid.uuid4()
            id = "CA" + str(xd)
            id = id[0:12]
        elif 2024 - year < 18 :
            xd = uuid.uuid4()
            id = "CC" + str(xd)
            id = id[0:12]

    def __get_name(self):
        return self.__name + " " + self.__last_name
    
    def __get_birthday(self):
        return self.__birthday
    
    def __get_curp(self):
        return self.__curp
    
    def __get_num_visits(self):
        return self.__num_visits
    
    def __get_date_of_register(self):
        return self.__date_of_register
    
    def show_client(self):
        
        print("\nID: %s   Full name: %s   Birthday: %s  Date of register: %s   CURP: %s   Number of visits %s\n", self.__get_id , self.__get_name, self.__get_birthday, self.__get_date_of_register, self.__get_curp, self.__get_num_visits)
        anyperson = anyperson + 1